profiler/src/ProfilerEngine/Directory.Build.targets

```diff
@@ -29,52 +29,44 @@
-          <DeployDirectoryBase Condition="'$(DeployToMonitoringHome)' != 'True'">$(BuildOutputRoot)\DDProf-Deploy</DeployDirectoryBase>
+          <TopLevelDeployDirectoryBase Condition="'$(DeployToMonitoringHome)' == 'True'">$(EnlistmentRoot)\shared\bin\monitoring-home</TopLevelDeployDirectoryBase>
+          <TopLevelDeployDirectoryBase Condition="'$(DeployToMonitoringHome)' != 'True'">$(BuildOutputRoot)\DDProf-Deploy</TopLevelDeployDirectoryBase>
+
+          <!-- In Linux case, we test against $(Platform), while on Windows case, we test against $(PlatformShortName). The reason is that for Windows, for   -->
+          <!-- 64bit, $(PlatformShortName) == to x64. But on Linux, $(PlatformShortName) == x86_64.                        -->
+          <DDPlatform Condition=" '$(PlatformShortName)' != '' AND $([MSBuild]::IsOSPlatform('Windows'))">win-$(PlatformShortName)</DDPlatform>
+          <DDPlatform Condition=" '$(PlatformShortName)' == '' AND $([MSBuild]::IsOSPlatform('Windows'))">win-unknown</DDPlatform>
+          <DDPlatform Condition=" '$(Platform)' != '' AND $([MSBuild]::IsOSPlatform('Linux')) ">linux-$(Platform)</DDPlatform>
+          <DDPlatform Condition=" '$(Platform)' == '' AND $([MSBuild]::IsOSPlatform('Linux')) ">linux-unknown</DDPlatform>
+          
+          <DeployDirectoryBase >$(TopLevelDeployDirectoryBase)\$(DDPlatform)</DeployDirectoryBase>
         </PropertyGroup>
 
         <PropertyGroup>
-            <!--
-            Note that the top-level Directory.Build.props defines a property "ConfigBasedRelativeOutputPath" that
-            combines $(Configuration) with the Platform moniker just like below. However we cannot use it, because
-            it uses the Platform moniker of the current build, whereas here we need all the different relevant Platform monikers.
-            So, if we ever renamed a platform moniker (it happened in the past), we would need to update the lines below accordingly.
-            -->
-            <NativeWindowsX86DeployFilesSrcDir>$(BaseOutputPath)\$(Configuration)-x86\profiler\src\ProfilerEngine\Datadog.Profiler.Native.Windows</NativeWindowsX86DeployFilesSrcDir>
-            <NativeWindowsX64DeployFilesSrcDir>$(BaseOutputPath)\$(Configuration)-x64\profiler\src\ProfilerEngine\Datadog.Profiler.Native.Windows</NativeWindowsX64DeployFilesSrcDir>
-
-            <NativeLinuxX86DeployFilesSrcDir>$(BaseOutputPath)\$(Configuration)-x86\profiler\src\ProfilerEngine\Datadog.Profiler.Native.Linux</NativeLinuxX86DeployFilesSrcDir>
-            <NativeLinuxX64DeployFilesSrcDir>$(BaseOutputPath)\$(Configuration)-x64\profiler\src\ProfilerEngine\Datadog.Profiler.Native.Linux</NativeLinuxX64DeployFilesSrcDir>
+            <NativeWindowsDeployFilesSrcDir>$(BaseOutputPath)\$(ConfigBasedRelativeOutputPath)\profiler\src\ProfilerEngine\Datadog.Profiler.Native.Windows</NativeWindowsDeployFilesSrcDir>
```

Comment:
> Removing the bitness info in the path might break here. Since x86 and x64 binaries have now the same name, they can be copied into the same folder: I don't know which one will erase the other.

I mean, yes, they _can_ be overwritten, but if they are, then we have a build problem 😄 They are currently (before _and_ after this change), built to separate folders, e.g. `Release-x86` and `Release-x64`.

When they're copied to DDProf-Deploy they're _deployed_ to the `win-x64` and `win-x86` folders, so there's again no issue with overwritting AFAICT?

There's a bigger question here for me - why do we need the `DDProf-Deploy` directory at all? For me, it adds confusion and complexity? It's not where things are _built_ to, so we could just "deploy" them to the Monitoring Home directory, where they end up eventually anyway?

I was a bit torn on the "dropping the bitness from the file names" question, but we ultimately opted for it for consistency, and to reduce the duplication in the arch-specific folders. That, coupled with the fact we _have_ to drop it in some cases for back compat reasons led us to this state... I think we can safely add it back in later though if it's an issue, it's only the `Datadog.Trace.ClrProfiler.Native` file we can't ever rename 🙂

Reply:
> When they're copied to DDProf-Deploy they're deployed to the win-x64 and win-x86 folders, so there's again no issue with overwritting AFAICT?
Where is this done?
The $(BaseOutputPath)\$(ConfigBasedRelativeOutputPath) does not seem to contain the bitness

Reply:
> Where is this done?
> The (ConfigBasedRelativeOutputPath) does not seem to contain the bitness

`ConfigBasedRelativeOutputPath` contains the bitness 🙂 It's (e.g.) `Release-x86` or `Release-x64`